var/spack/repos/builtin/packages/mimalloc/package.py

```diff
@@ -12,12 +12,45 @@ class Mimalloc(CMakePackage):
     homepage = "https://microsoft.github.io/mimalloc"
     url = "https://github.com/microsoft/mimalloc/archive/v0.0.0.tar.gz"
     git = "https://github.com/microsoft/mimalloc.git"
-    maintainers = ['msimberg']
+    maintainers = ["msimberg"]
 
-    version('dev-slice', branch='dev-slice')
-    version('dev', branch='dev')
-    version('master', branch='master')
-    version('2.0.6', sha256='9f05c94cc2b017ed13698834ac2a3567b6339a8bde27640df5a1581d49d05ce5')
-    version('1.7.6', sha256='d74f86ada2329016068bc5a243268f1f555edd620b6a7d6ce89295e7d6cf18da')
+    version("dev-slice", branch="dev-slice")
+    version("dev", branch="dev")
+    version("master", branch="master")
+    version("2.0.6", sha256="9f05c94cc2b017ed13698834ac2a3567b6339a8bde27640df5a1581d49d05ce5")
+    version("1.7.6", sha256="d74f86ada2329016068bc5a243268f1f555edd620b6a7d6ce89295e7d6cf18da")
 
-    depends_on('cmake@3.0:', type='build')
+    depends_on("cmake@3.0:", type="build")
+
+    mimalloc_options = {
+        "secure": (False, "Use full security mitigations (like guard pages, allocation randomization, double-free mitigation, and free-list corruption detection)", None),
+        "debug_full": (False, "Use full internal heap invariant checking in DEBUG mode (expensive)", None),
+        "padding": (True, "Enable padding to detect heap block overflow (used only in DEBUG mode)", None),
+        "override": (True, "Override the standard malloc interface (e.g. define entry points for malloc() etc)", None),
+        "xmalloc": (False, "Enable abort() call on memory allocation failure by default", None),
+        "show_errors": (False, "Show error and warning messages by default (only enabled by default in DEBUG mode)", None),
+        "use_cxx": (False, "Use the C++ compiler to compile the library (instead of the C compiler)", None),
+        "see_asm": (False, "Generate assembly files", None),
+        "osx_interpose": (True, "Use interpose to override standard malloc on macOS", "platform=darwin"),
+        "osx_zone": (True, "Use malloc zone to override standard malloc on macOS", "platform=darwin"),
+        "local_dynamic_tls": (False, "Use slightly slower, dlopen-compatible TLS mechanism (Unix)", None),
+        "build_shared": (True, "Build shared library", None),
+        "build_static": (True, "Build static library", None),
+        "build_object": (True, "Build object library", None),
```

Comment:
Can we have a `libs` multi-valued variant, like done in many other packages?

Reply:
Yeah, I don't see why not. I was mostly being lazy and doig the straightforward mapping. I'll update this.

Reply:
Thanks

Reply:
I've updated this.